consoleme/lib/aws.py

```diff
@@ -1262,3 +1274,172 @@ async def validate_iam_policy(policy: str, log_data: Dict):
         policy, log_data, policy_type="IDENTITY_POLICY"
     )
     return parliament_findings + access_analyzer_findings
+
+
+async def get_all_scps(force_sync=False) -> Dict[str, ServiceControlPolicyModel]:
+    """Retrieve a dictionary containing all Service Control Policies across organizations
+
+    Args:
+        force_sync: force a cache update
+    """
+    redis_key = config.get(
+        "cache_scps_across_organizations.redis.key.all_scps_key", "ALL_AWS_SCPS"
+    )
+    scps = await retrieve_json_data_from_redis_or_s3(
+        redis_key,
+        s3_bucket=config.get("cache_scps_across_organizations.s3.bucket"),
+        s3_key=config.get("cache_scps_across_organizations.s3.file"),
+        default={},
+    )
+    if force_sync or not scps:
+        scps = await cache_all_scps()
+    scp_models = {}
+    for account, org_scps in scps.items():
+        scp_models[account] = [ServiceControlPolicyModel(**scp) for scp in org_scps]
+    return scp_models
+
+
+async def cache_all_scps() -> Dict[str, Any]:
+    """Store a dictionary of all Service Control Policies across organizations in the cache"""
+    all_scps = {}
+    for organization in config.get("cache_scps_across_organizations.organizations", []):
```

Comment:
Ditto here, maybe we should have a config struct that we share for a list of orgs. Granted, there may be a case where you want to cache some details about an org but not SCPs, or visa versa. Perhaps we should draw the line and not support those use cases